### ISSUE TITLE
Allow reflectAll to accept an object of functions

### ISSUE THREAD

**Issue by gpkeene** [caolan/async#1195]
Currently, the `reflectAll` function only accepts an array of functions and `reflect` accepts single functions.  It would be a nice quality of life update to have `reflectAll` accept objects containing functions similar to other functions like `series` and `parallel`.  This would allow anyone who has code similar to below to simply wrap their existing object with `reflectAll` rather than wrap each function with `reflect`.  The results could then be accessed by `results.one.value` or `results.three.error`, just as if each function had been wrapped individually.
```
async.series({
    one: function(callback) {
        callback(null, 'one');
    },
    two: function(callback) {
        callback(null, 'two');
    },
    three: function(callback) {
        callback('errorerrorerror');
    },
    four: function(callback) {
        callback(null, 'four');
    },
    five: function(callback) {
        callback(null, 'five');
    }
}, function(err, results) {
    console.log('errors',err);
    console.log('results',results);
});

```

**Comment by aearly** [comment]
A PR would be welcome. :)

**Comment by aearly** [comment]
Closed by #1207
